### PR TITLE
feature/jwt-ext-auth

### PR DESCRIPTION
**Pull request by EsamSharaf**
Implementing Flask-JWT-Extended extension for authentication

### CODE REVIEW

File: anime/views.py [hunk 1]
```diff
@@ -113,3 +119,36 @@ def create_anime():
...
 
     return anime_schema.dump(schema_dict), 201
+
+
+@animes_bp.route('/api/v1/register', methods=['POST'])
+def register():
+    username = request.json.get("username")
+    password = request.json.get("password")
+
+    user = User.query.filter_by(username=username).one_or_none()
```
Comment (blancheta) [inline anime/views.py:129]: We should check the user schema before checking types before looking for the user in the database.

File: anime/views.py [hunk 2]
```diff
@@ -113,3 +119,36 @@ def create_anime():
...
+    db.session.add(user)
+    db.session.commit()
+    return jsonify(message='user created'), 201
+
+
+@animes_bp.route('/api/v1/login', methods=['POST'])
+def login():
+    username = request.json.get("username")
+    password = request.json.get("password")
+
```
Comment (blancheta) [inline anime/views.py:146]: Same here. We should check the user schema before checking types before looking for the user in the database.

**Review by blancheta — COMMENTED** [review]
Looks good to me.

Small question. How do you understand the difference between 401 and 403?

**Comment by EsamSharaf** [comment]
[about this comment](https://github.com/EsamSharaf/anime_API/pull/9#pullrequestreview-1977170415), as far as I know that 401 is Unauthorized status code, while 403 Forbidden. I saw some use them interchangeably.

File: anime/views.py [hunk 3]
```diff
@@ -113,3 +121,52 @@ def create_anime():
     db.session.commit()
 
     return anime_schema.dump(schema_dict), 201
+
+
+@animes_bp.route('/api/v1/register', methods=['POST'])
+def register():
+    """Route for creating a new user
```
Comment (blancheta) [inline anime/views.py:128]: Instead of a route, you can call that View or Handler

**Review by blancheta — APPROVED** [review]

**Pull request merged**